### PR TITLE
Controllers

### PR DESCRIPTION
**Pull request by americool**
deals with issues #4, #5, #6 as well as potentially sends viable JSON to front end for randomly generated anecdote as well as count of various illnesses. Ryan may be working on seeing sample data so we can better test this, but I thought I'd put it up.

### CODE REVIEW

File: app/controllers/api/v1/research_controller.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+module Api
+  module V1
+    class ResearchController < ActionController::API
+
+      def create
+        @research = Research.create(research_params)
+      end
+
+      private
+      def submission_params
```
Comment (brittanyjspringer) [inline app/controllers/api/v1/research_controller.rb]: @americool I think this is a typo: should be research_params.

File: app/controllers/api/v1/anecdote_controller.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    def index
+    end
+
+    def create
+      @anecdote  = Anecdote.create(anecdote_params)
+    end
+
+    def random
+      # JSON payload: Frontend will send you an illness, return a single anecdote that matches that illness
+      random = Anecdote.all.includes(:submission).where(submissions: {illness: params[:illness]}).sample
```
Comment (brittanyjspringer) [inline app/controllers/api/v1/anecdote_controller.rb]: Glad that logic worked out!

**Review by brittanyjspringer — CHANGES_REQUESTED** [review]

**Comment by brittanyjspringer** [comment]
@americool There is a typo that needs to be fixed. Can you also verify that when you hit the API from the new controllers that login is required? It was letting me through without it. I suspect you will need to add 'before_action :authenticated?' on the these controllers. Thanks!

File: app/controllers/api/v1/research_controller.rb (hunk 1 above)
Comment (ryan-on-rails) [reply app/controllers/api/v1/research_controller.rb]: Also, I just realized that the controller filename and declaration need to be the plural of the model so the filename and controller name should be changed to `researches_controller.rb` & `ResearchesController` respectively.

**Review by ryan-on-rails — CHANGES_REQUESTED** [review]
The `research_controller.rb` needs to be updated to reflect the plural of the `research` model.

**Review by ryan-on-rails — APPROVED** [review]

**Comment by americool** [comment]
Have we dealt with the authentication issue? Little confused as to why were updating research to researches - otherwise it seems good to me.

**Comment by ryan-on-rails** [comment]
I think we still need to do some tweaking for the authentication because I'm playing around with it now and can't seem to get a request through yet. Also, the `researches` thing is just a rails naming convention. We could bypass it if we want but I figured it was easier to conform at this point.

**Review by brittanyjspringer — APPROVED** [review]

**Pull request merged**